feat: Add Hummingbot plugin

SumeetChougule opened the pull request:
# Relates to:
No specific issue - This is a new plugin contribution

# Risks
Low - This plugin is self-contained within its own package and doesn't modify any existing functionality. The only integration point is through the plugin system which is designed for extensibility.

# Background

## What does this PR do?
Adds a new Hummingbot plugin that enables Eliza agents to perform automated market making and trading operations. The plugin provides:
- Real-time market data streaming via WebSocket
- Simple market making strategy with configurable parameters
- Order lifecycle management (create, cancel, track)
- Inventory skew management
- Type-safe configuration and error handling

## What kind of change is this?
Features (non-breaking change which adds functionality)

# Documentation changes needed?
My changes do not require a change to the project documentation as the plugin includes its own comprehensive README.md with installation, configuration, and usage instructions.

# Testing

## Where should a reviewer start?
1. Review the plugin's architecture in `packages/plugin-hummingbot/src/index.ts`
2. Check the market making strategy implementation in `packages/plugin-hummingbot/src/strategies/simple-market-making.ts`
3. Review type definitions and configuration interface in `packages/plugin-hummingbot/src/types.ts`

## Detailed testing steps
1. Install dependencies with `pnpm install`
2. Configure Hummingbot instance details in your character config:
```json
{
  "plugins": ["@eliza/plugin-hummingbot"],
  "settings": {
    "HUMMINGBOT_CONFIG": {
      "instance": {
        "url": "http://localhost:15888",
        "apiKey": "your-api-key"
      }
    }
  }
}

odilitime committed:
isFalsish() for parsing environment in a standard way

odilitime committed:
roll isFalsish into parseBooleanFromText

odilitime committed:
empty is already mapped to null

odilitime committed:
move all env parsing here

odilitime committed:
add twitterConfig cstr param and prefer it over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSettings

odilitime committed:
pass twitterConfig as 2nd parameter to cstr, only start search once

mcp97 committed:
Merge branch 'develop' into fix/typescript-config

mcp97 committed:
Merge pull request #1485 from ShaneOxM/fix/typescript-config

fix: add required incremental option and remove invalid typescript configuration

mcp97 committed:
Merge branch 'develop' into 1506--clean-cache

odilitime committed:
fix TWITTER_POLL_INTERVAL scale

odilitime committed:
fix TWITTER_POLL_INTERVAL scale

odilitime committed:
Merge branch 'develop' into test-eliza

odilitime committed:
make MAX_TWEET_LENGTH integer since that's how it's consomed/compared

odilitime committed:
make MAX_TWEET_LENGTH integer since that's how it's consomed/compared

odilitime committed:
Merge branch 'test-eliza' of https://github.com/odilitime/eliza into test-eliza

mcp97 committed:
Merge pull request #1508 from timolegros/1506--clean-cache

Clear `/cache/` in `clean.sh` script

shakkernerd committed:
Revert "Clear `/cache/` in `clean.sh` script"

shakkernerd committed:
Merge pull request #1515 from elizaOS/revert-1508-1506--clean-cache

chore: Revert "Clear `/cache/` in `clean.sh` script"

shakkernerd committed:
chore: remove cache in core

shakkernerd committed:
Merge pull request #1516 from elizaOS/fix/remove_cache

chore: remove cache in core

shakkernerd committed:
Merge branch 'develop' into add-fereProPlugin

shakkernerd committed:
chore: remove package-lock file

shakkernerd committed:
chore: remove empty README file

shakkernerd committed:
feat: add eslint support

shakkernerd committed:
chore: remove unused imports

shakkernerd committed:
chore: remove typescript from package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: add npmignore file

shakkernerd committed:
Merge branch 'develop' into test-eliza

odilitime committed:
Merge branch 'develop' into patch-1

shakkernerd committed:
revert: packages/client-github/src/index.ts to match develop

odilitime committed:
update org

odilitime committed:
Merge pull request #1312 from y4my4my4m/patch-1

fix: Update speech.ts

odilitime committed:
Merge branch 'develop' into fix

odilitime committed:
Merge pull request #1456 from pythonberg1997/fix

fix: swap and bridge actions of plugin-evm

odilitime committed:
Merge branch 'develop' into test-eliza

oxSaturn committed:
fix: handle long tweet in utils

imtms committed:
feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

imtms committed:
feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

odilitime committed:
Merge branch 'develop' into fix/long-tweet

shakkernerd committed:
Merge branch 'develop' into pr-1514

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

Titan-Node committed:
add Livepeer configuration to .env.example

Titan-Node committed:
add Livepeer gateway URL to agent configuration

Titan-Node committed:
add Livepeer model configuration

Titan-Node committed:
add Livepeer model and provider name to core types

shakkernerd committed:
Merge branch 'develop' into test-eliza

Titan-Node committed:
Included image generation action to include validation for Livepeer gateway URL

Titan-Node committed:
Implement Livepeer image generation logic with URL validation

shakkernerd committed:
Merge pull request #1514 from odilitime/test-eliza

fix: client-twitter lowerCase bug and environment clean up (+lint fixes, and TWITTER_SEARCH_ENABLE double start fix)

tsubasakong committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge branch 'develop' into develop

samarth30 committed:
fix: fixed dependecny issue and resolved merge conflicts

shakkernerd committed:
Merge pull request #1522 from imtms/develop

feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

odilitime committed:
lint: remove unused error variable

odilitime committed:
lint: fix EmbeddingProvider already defined error (via claude)

odilitime committed:
prefer client.twitterConfig over getSetting

odilitime committed:
Merge branch 'develop' into fix/long-tweet

odilitime committed:
Merge pull request #1520 from oxSaturn/fix/long-tweet

fix: handle long tweet in utils

Titan-Node committed:
Add Livepeer model and configuration details to documentation

- Updated ModelProviderName.md to include Livepeer as a model provider.
- Added Livepeer model alias in Models.md.
- Included Livepeer gateway URL in quickstart.md and configuration.md.
- Enhanced documentation to guide users on setting up Livepeer models.

vanshika-srivastava committed:
Update package.json with build-docker command

HowieDuhzit committed:
feat: add /:agentId/speak endpoint for text-to-speech functionality

- Implemented a new POST endpoint that allows users to send text to a specified agent.
- The endpoint processes the message, generates a response, and converts it to speech using the ElevenLabs API.
- Added error handling for missing text, agent not found, and ElevenLabs API errors.
- Responses are streamed back as audio in MPEG format.

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #1528 from HowieDuhzit/main

feat: add /:agentId/speak endpoint for text-to-speech functionality

odilitime committed:
Merge branch 'develop' into add-livepeer-image-gen-1

vanshika-srivastava committed:
Merge branch 'develop' into main

shakkernerd committed:
chore: update package.json

shakkernerd committed:
Merge pull request #1527 from vanshika-srivastava/main

Fix: Update package.json with build-docker command to match the dockerfile command

odilitime committed:
Merge branch 'develop' into add-livepeer-image-gen-1

odilitime committed:
Merge pull request #1525 from Titan-Node/add-livepeer-image-gen-1

feat: Add Livepeer Image Provider

shakkernerd committed:
Merge branch 'develop' into main

odilitime committed:
remove unused var

shakkernerd committed:
Merge pull request #839 from tsubasakong/main

feat: Add Custom System Prompt Support for plugin-image-generation

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

odilitime committed:
remove unused vars per lint

shakkernerd committed:
chore: remove unused imports

odilitime committed:
Merge branch 'develop' into odi-fix

shakkernerd committed:
fix: unused variable

shakkernerd committed:
fix: unused variable

odilitime committed:
Merge pull request #1529 from elizaOS/odi-fix

chore: remove unused vars

shakkernerd committed:
fix: unused variable

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' into pr-1126

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1530 from elizaOS/new_version_0.1.7-alpha.2

chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1519 from elizaOS/develop

chore: 0.1.7-alpha.2 prep, develop => main

actions-user committed:
chore: update changelog

e-fu committed:
fix(client-twitter): tighten engagement criteria and raise quality thresholds

e-fu committed:
Merge branch 'main' into fix/twitter-engagement-criteria

SK1989sL committed:
Update README.md

RedHorse823 committed:
Update README_CN - Model Configuration

The original model configuration guide is misleading, it tells readers to configure model provider by change the `XAI_MODEL` parameters.

But this is for the XAI configurations, i update the description completely with the right configuration steps.

swizzmagik committed:
Merge remote-tracking branch 'upstream/Main'

yanushevitz committed:
Added Polish Readme file

yanushevitz committed:
Fixed polish's readme typo's

yanushevitz committed:
Rename Readme_PL.md to README_PL.md

swizzmagik committed:
feat: fix chat client to autoscroll

swizzmagik committed:
chore: update changelog

mcp97 committed:
Merge pull request #1538 from swizzmagik/feat/add-autoscroll-chat-client

mcp97 committed:
Merge branch 'develop' into Polish_readme

mcp97 committed:
Merge pull request #1537 from yanushevitz/Polish_readme

Polish readme

mcp97 committed:
Merge branch 'develop' into Update-ReadMe-CN

mcp97 committed:
Merge pull request #1535 from RedHorse823/Update-ReadMe-CN

Update README_CN - Model Configuration

mcp97 committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge branch 'develop' into fix/twitter-engagement-criteria

mcp97 committed:
Merge pull request #1534 from SK1989sL/patch-1

docs: fix grammar/typos in README

mcp97 committed:
feat: improve messages

Improve message so criteria is displayed

mcp97 committed:
Merge branch 'develop' into fix/twitter-engagement-criteria

mcp97 committed:
Merge branch 'develop' into realitySpiral/dexExp

mcp97 committed:
Merge pull request #1533 from e-fu/fix/twitter-engagement-criteria

Fix: make twitter engagement criteria in prompt more focused on relevance to topics/interests/character

mcp97 committed:
Merge branch 'develop' into realitySpiral/dexExp

odilitime committed:
fix number prefixed username, add X brand in messaging

odilitime committed:
Merge branch 'develop' into twitter-fix

odilitime committed:
Merge pull request #1540 from elizaOS/realitySpiral/dexExp

feat: improve stale pr/issue messages

odilitime committed:
Merge branch 'develop' into twitter-fix

odilitime committed:
Merge pull request #1541 from odilitime/twitter-fix

fix: twitter usernames can start with numbers

mgunnin committed:
refactor: reorganize imports in video service

mgunnin committed:
fix: improve clarity and grammar in pull request template

azep-ninja committed:
fix tg markdown issue.

Ed-Marcavage committed:
fix: update jsdoc automation workflow

Ed-Marcavage committed:
fix: update jsdoc automation workflow

Ed-Marcavage committed:
Merge branch 'develop' into fix/jsdoc-automation

madjin committed:
Merge pull request #1547 from Ed-Marcavage/fix/jsdoc-automation

Fix jsdoc automation build issue & parametrize PR branch target

shakkernerd committed:
Merge branch 'develop' into fix--Telegram-Markdown-issues

madjin committed:
docs: Add JSDoc documentation to constants.ts

madjin committed:
docs: Add JSDoc documentation to createRuntime.ts

madjin committed:
docs: Add JSDoc documentation to testSetup.ts

madjin committed:
docs: Add JSDoc documentation to types.ts

mcp97 committed:
Merge pull request #1546 from azep-ninja/fix--Telegram-Markdown-issues

fix: Telegram Bad Request: can't parse entities

mcp97 committed:
Merge branch 'develop' into code-cleanup

madjin committed:
docs: Add JSDoc documentation to constants.ts

madjin committed:
docs: Add JSDoc documentation to createRuntime.ts

madjin committed:
docs: Add JSDoc documentation to types.ts

madjin committed:
docs: Add JSDoc documentation to testSetup.ts

shakkernerd committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1545 from mgunnin/code-cleanup

Chore: Refactor Imports. Improve PR Template

madjin committed:
Merge branch 'develop' into docs-update-full-1735429229223

mcp97 committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1549 from elizaOS/docs-update-full-1735429229223

docs: Add JSDoc documentation

mcp97 committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1548 from elizaOS/docs-update-full-1735428826331

docs: Add JSDoc documentation

HashWarlock committed:
add character file script to make system prompt and templates easier to add to existing character files or generate a new one

zkfriendly committed:
feat: add theme toggle functionality with dark and light mode support

- Imported ThemeToggle component in AppSidebar
- Added ThemeToggle to SidebarFooter in AppSidebar
- Initialized theme based on localStorage and system preferences in main.tsx to support dark and light modes

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1555 from zkfriendly/main

feat: add theme toggle functionality with dark and light mode support

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 7" notes

mcp97 committed:
Merge pull request #1559 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #7" notes

shakkernerd committed:
Merge branch 'develop' into pr-1554

Commit:
Add Hummingbot plugin for market making

This commit adds a new plugin that integrates Hummingbot's market making capabilities with Eliza trading agents. The plugin provides:
- Real-time market data streaming
- Simple market making strategy with configurable parameters
- Order lifecycle management
- Inventory skew management

samwhosung committed:
add TEE plugin-env support

samwhosung committed:
fix

samwhosung committed:
fix

samarth30 reviewed (changes requested):
Hi @SumeetChougule ,

Could you push your changes after taking the latest pull to ensure everything is up to date?

Also, it would be great if you could replace references to @ai16z with @elizaos and update @elizaos/eliza to @elizaos/core to avoid any potential errors in the code.

Additionally, could you integrate your Hummingbot plugin into agent/src/index.ts, following the structure of other plugins? Lastly, don’t forget to include your .env configurations in the .env.example file for consistency.

L-jasmine committed:
fix: init 768 dimension in database for gaianet

Commit:
refactor: Update package references and integrate Hummingbot plugin

- Replace @ai16z/eliza with @elizaos/core
- Add Hummingbot configuration to .env.example
- Integrate Hummingbot plugin into agent/src/index.ts
- Export hummingbotPlugin function

Commit:
docs: Update Hummingbot plugin configuration for consistency

- Update environment variable references to use HUMMINGBOT_API_KEY consistently
- Add Environment Variables section to README
- Use environment variable references in configuration examples

shakkernerd committed:
chore: parse files through prettier

shakkernerd committed:
Merge pull request #1573 from elizaOS/enhance/update_script

feat: parse files through prettier

shakkernerd committed:
Merge branch 'pr-1572' into develop

tomguluson92 committed:
Update README.md

tomguluson92 committed:
Add files via upload

odilitime committed:
Merge branch 'develop' into patch-4

odilitime committed:
Merge pull request #1574 from tomguluson92/patch-4

docs: overview and preview of technical report

odilitime committed:
Merge branch 'develop' into samuv/env-tee

odilitime committed:
Merge pull request #1571 from samuveth/samuv/env-tee

feat: add TEE support for plugin-env

yorkerhodes3 committed:
Update agents.md

Minor update to correct "should"

treppers committed:
add an client-direct endpoint to get memories by agentid and roomid

mcp97 committed:
Merge branch 'develop' into patch-1

zhourunlai committed:
update volcengine model

twilwa committed:
Merge branch 'develop' into memories_endpoint

0xFloyd committed:
docs: fix Contributing Guide

chandiniv1 committed:
feat: add docs for image generation plugin

salazarsebas committed:
fix: fix the image redirection

salazarsebas committed:
feat: Add Spanish Translation for Documentation README

mcp97 committed:
Merge pull request #1579 from yorkerhodes3/patch-1

Update agents.md - minor update

mcp97 committed:
Merge branch 'develop' into cleanup/documentation

ileana-pr committed:
added README files to all plugins

zhourunlai committed:
use tavily sdk

ileana-pr committed:
file updates

mcp97 committed:
Merge branch 'develop' into feat/add-readme-spanish-version-in-docs

bendanzhentan committed:
refactor(plugin-conflux): output detailed invalid content

zkvm committed:
tweak of evm transfer template

zhourunlai committed:
add experimental telemetry model option

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge remote-tracking branch 'origin/develop' into pr/bendanzhentan/1602

Commit:
chore: Update pnpm lockfile

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1602 from bendanzhentan/refactor-output-conflux-errors

refactor(plugin-conflux): output detailed invalid content

shakkernerd committed:
Merge branch 'develop' into fix-plugin-evm-transfer-action

shakkernerd committed:
Merge pull request #1604 from zkvm/fix-plugin-evm-transfer-action

fix: tweak transfer template of plugin-evm

shakkernerd committed:
Merge remote-tracking branch 'origin/develop' into pr/zhourunlai/1603

shakkernerd committed:
Merge pull request #1603 from zhourunlai/experimental_telemetry

feat: add experimental telemetry model option

shakkernerd committed:
Merge branch 'develop' into readme-plugin-files

shakkernerd committed:
Merge pull request #1601 from ileana-pr/readme-plugin-files

docs: README.md files for plugins

shakkernerd committed:
chore: fix conflicts

shakkernerd committed:
chore: add readme

shakkernerd committed:
fix: set publishedDate to optional as stated by the docs.

shakkernerd committed:
chore: move tavily package to the core

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
feat: add error handling for TAVILY_API_KEY key

shakkernerd committed:
Merge branch 'develop' into tavily_sdk

shakkernerd committed:
Merge pull request #1599 from zhourunlai/tavily_sdk

feat: use tavily sdk

shakkernerd committed:
Merge branch 'develop' into feat/add-readme-spanish-version-in-docs

shakkernerd committed:
Merge pull request #1594 from salazarsebas/feat/add-readme-spanish-version-in-docs

docs: add readme spanish version in docs

shakkernerd committed:
Merge branch 'develop' into chandini/image-generaration-plugin

shakkernerd committed:
Merge pull request #1591 from chandiniv1/chandini/image-generaration-plugin

feat: add docs for image generation plugin

shakkernerd committed:
chore: remove conflicting files

shakkernerd committed:
chore: README files

shakkernerd committed:
chore: Update local-development.md

shakkernerd committed:
Merge branch 'develop' into cleanup/documentation

shakkernerd committed:
Merge pull request #1589 from 0xFloyd/cleanup/documentation

docs: fix Contributing Guide

shakkernerd committed:
Merge branch 'develop' into volcengine_model

shakkernerd committed:
Merge pull request #1586 from zhourunlai/volcengine_model

feat: update volcengine model

azep-ninja committed:
fix continue double responses

azep-ninja committed:
fix google api key

shakkernerd committed:
feat: include priority to plugin-tee build

shakkernerd committed:
chore: remove build cache with clean command

shakkernerd committed:
chore: changes to settings

shakkernerd committed:
fix: add @tavily/core to external libs (almost took my life)

shakkernerd committed:
chore: remove logs

shakkernerd committed:
chore: console -> elizaLogger

Commit:
fix: update cancelAllOrders to handle orders without id

Updated the cancelAllOrders function to handle orders that might not have an id property by falling back to clientOrderId

shakkernerd committed:
Merge pull request #1609 from elizaOS/fix/breath_taking_error

fix: breath taking error

shakkernerd committed:
Merge branch 'develop' into fix/google-api-key-character-file

Commit:
ci: update workflow to clean build artifacts

- Add step to clean dist and tsbuildinfo files before build
- Add caching for pnpm store
- Ensure clean installation of dependencies

shakkernerd committed:
Merge pull request #1607 from azep-ninja/fix/google-api-key-character-file

fix: Google API Key not passing from character file

azep-ninja committed:
Merge branch 'develop' into fix/continue-double-response

cole-gillespie committed:
add plugins to the key components section of the faq

during a recent discussion on discord it was not clear to some folks the conceptual difference between clients and plugins. this just adds a bit more high level info up top on the FAQ for future creators.

cole-gillespie committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1614 from cole-gillespie/patch-1

add plugins to the key components section of the faq

mcp97 committed:
Merge branch 'develop' into memories_endpoint

mcp97 committed:
Merge branch 'develop' into fix/continue-double-response

mcp97 committed:
Merge pull request #1581 from treppers/memories_endpoint

add an client-direct endpoint to get memories by agentid and roomid

azep-ninja committed:
Merge branch 'develop' into fix/continue-double-response

azep-ninja committed:
re-build fx

mcp97 committed:
Merge pull request #1606 from azep-ninja/fix/continue-double-response

fix: Double Responses from Continue Action

SumeetChougule committed:
Merge branch 'develop' into plugin-hummingbot

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.
